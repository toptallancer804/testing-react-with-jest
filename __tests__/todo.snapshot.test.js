import Todo from '../app/todo';
import renderer from 'react-test-renderer';
import React from 'react';

describe('Todo component', () => {
  describe('Rendering the todo', () => {
    it('renders correctly', () => {
      const todo = { id: 1, done: false, name: 'Buy Milk' };
      const rendered = renderer.create(
        <Todo todo={todo} />
      );
      expect(rendered.toJSON()).toMatchSnapshot();
    });
  });

  // describe('toggling the todo', () => {
  //   it('calls the given doneChange callback prop', () => {
  //     const todo = { id: 1, done: false, name: 'Buy Milk' };
  //     const doneChange = jest.fn();
  //     const rendered = TestUtils.renderIntoDocument(
  //       <Todo todo={todo} doneChange={doneChange} />
  //     );

  //     const todoP = TestUtils.findRenderedDOMComponentWithClass(rendered, 'toggle-todo');
  //     TestUtils.Simulate.click(todoP);
  //     expect(doneChange).toBeCalledWith(1);
  //   });
  // });
});
