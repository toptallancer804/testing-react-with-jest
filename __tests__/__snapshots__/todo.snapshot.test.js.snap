exports[`Todo component Rendering the todo renders correctly 1`] = `
<div
  className="todo  todo-1">
  <p
    className="toggle-todo"
    onClick={[Function onClick]}>
    Buy Milk
  </p>
  <a
    className="delete-todo"
    href="#"
    onClick={[Function onClick]}>
    Delete
  </a>
</div>
`;
